packages/firestore/test/unit/index/firestore_index_value_writer.test.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0x00 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0x00
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { expect } from 'chai';
+
+import { FirestoreIndexValueWriter } from '../../../src/index/firestore_index_value_writer';
+import { IndexByteEncoder } from '../../../src/index/index_byte_encoder';
+import { Timestamp } from '../../../src/lite-api/timestamp';
+import { IndexKind } from '../../../src/model/field_index';
+import type { Value } from '../../../src/protos/firestore_proto_api';
+import { toTimestamp } from '../../../src/remote/serializer';
+import { JSON_SERIALIZER } from '../local/persistence_test_helpers';
+
+describe('Firestore Index Value Writer', () => {
```

Comment:
We should also have some tests to make sure the ordering of the indexed values, in addition to check equality.